shared_model/builders/protobuf/builder_templates/transaction_template.hpp

```diff
@@ -0,0 +1,334 @@
+          command->set_account_id(account_id);
+          command->set_public_key(crypto::toBinaryString(public_key));
+        });
+      }
+
+      auto removeSignatory(
+          const interface::types::AccountIdType &account_id,
+          const interface::types::PubkeyType &public_key) const {
+        return addCommand([&](auto proto_command) {
+          auto command = proto_command->mutable_remove_sign();
+          command->set_account_id(account_id);
+          command->set_public_key(crypto::toBinaryString(public_key));
+        });
+      }
+
+      auto appendRole(const interface::types::AccountIdType &account_id,
+                      const interface::types::RoleIdType &role_name) const {
+        return addCommand([&](auto proto_command) {
+          auto command = proto_command->mutable_append_role();
+          command->set_account_id(account_id);
+          command->set_role_name(role_name);
+        });
+      }
+
+      auto createAsset(const interface::types::AssetNameType &asset_name,
+                       const interface::types::DomainIdType &domain_id,
+                       interface::types::PrecisionType precision) const {
+        return addCommand([&](auto proto_command) {
+          auto command = proto_command->mutable_create_asset();
+          command->set_asset_name(asset_name);
+          command->set_domain_id(domain_id);
+          command->set_precision(precision);
+        });
+      }
+
+      auto createAccount(
+          const interface::types::AccountNameType &account_name,
+          const interface::types::DomainIdType &domain_id,
+          const interface::types::PubkeyType &main_pubkey) const {
+        return addCommand([&](auto proto_command) {
+          auto command = proto_command->mutable_create_account();
+          command->set_account_name(account_name);
+          command->set_domain_id(domain_id);
+          command->set_main_pubkey(crypto::toBinaryString(main_pubkey));
+        });
+      }
+
+      auto createDomain(
+          const interface::types::DomainIdType &domain_id,
+          const interface::types::RoleIdType &default_role) const {
+        return addCommand([&](auto proto_command) {
+          auto command = proto_command->mutable_create_domain();
+          command->set_domain_id(domain_id);
+          command->set_default_role(default_role);
+        });
+      }
+
+      auto createRole(
+          const interface::types::RoleIdType &role_name,
+          std::initializer_list<interface::types::PermissionNameType>
+              permissions) const {
+        return createRole(role_name, permissions);
+      }
+
+      template <typename... Permission>
+      auto createRole(const interface::types::RoleIdType &role_name,
+                      const Permission &... permissions) const {
+        return createRole(role_name, {permissions...});
+      }
+
+      template <typename Collection>
+      auto createRole(const interface::types::RoleIdType &role_name,
```

Comment:
This method should be moved above other two `createRole`, because they use this method.